contracts/examples/erc721-transfer/FxERC721ChildTunnel.sol

```diff
@@ -97,16 +102,42 @@ contract FxERC721ChildTunnel is FxBaseChildTunnel, Create2, IERC721Receiver {
     }
 
     function _syncDeposit(bytes memory syncData) internal {
-        (address rootToken, address depositor, address to, uint256 tokenId, bytes memory depositData) = abi.decode(syncData, (address, address, address, uint256, bytes));
+        (address rootToken, address depositor, address to, uint256 tokenId, bytes memory depositData) = abi.decode(
+            syncData,
+            (address, address, address, uint256, bytes)
+        );
         address childToken = rootToChildToken[rootToken];
 
         // deposit tokens
         IFxERC721 childTokenContract = IFxERC721(childToken);
         childTokenContract.mint(to, tokenId, depositData);
     }
 
+    function _withdraw(
+        address childToken,
+        address receiver,
+        uint256 tokenId,
+        bytes memory data
+    ) internal {
+        IFxERC721 childTokenContract = IFxERC721(childToken);
+        // child token contract will have root token
+        address rootToken = childTokenContract.connectedToken();
+
+        // validate root and child token mapping
+        require(
+            childToken != address(0x0) && rootToken != address(0x0) && childToken == rootToChildToken[rootToken],
+            "FxERC721ChildTunnel: NO_MAPPED_TOKEN"
+        );
+
+        // withdraw tokens
+        childTokenContract.burn(tokenId);
```

Comment:
childTokenContract uses msg.sender internally? Other tokens uses user's address to burn them, why does this contract uses the tunnel itself?

Reply:
We are using OpenZeppelin implementations of the ERC token contracts, in the case of `ERC721` there is no such parameter, hence we don't have it either.